cpp/include/cuml/fil/multi_reduction.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+#include <raft/cuda_utils.cuh>
+// data [T]ype, reduction [R]adix
+template <int R, typename T>
+/**
+ per_thread holds one value per thread in shared memory
+ n_groups is the number of indendent reductions
+ per_thread values are ordered such that the stride is 1 for values belonging
+ to the same group and n_groups for values that are to be added together
+ n_values is the size of each individual reduction,
+ that is the number of values to be reduced to a single value
+*/
+__device__ T multi_reduction(T* data, int n_groups, int n_values) {
```

Comment:
This is actually just a sum (not a reduction with an arbitrary binary operation). Consider renaming it to `multi_reduce_sum` or something like that.

It would also be nice to convert it to a reduction with an arbitrary operation, but in another pull request.